src/backend/app/projects/image_processing.py

```diff
@@ -0,0 +1,79 @@
+import os
+import time
+import requests
+from loguru import logger as log
+
+
+class TaskProcessor:
+    def __init__(self, token, task_name, images_dir, base_url):
+        self.base_url = base_url
+        self.token = token
+        self.task_name = task_name
+        self.images_dir = images_dir
+        self.task_uuid = None
+
+    def init_task(self):
+        url = f"{self.base_url}/task/new/init"
+        payload = {
+            "name": self.task_name,
+            "webhook": "",
+            "skipPostProcessing": True,
+            "options": [],
+        }
+        response = requests.post(url, params={"token": self.token}, json=payload)
+        response.raise_for_status()
+        self.task_uuid = response.json()["uuid"]
+        log.info(f"Task initialized with UUID: {self.task_uuid}")
+
+    def upload_images(self):
+        for image_file in os.listdir(self.images_dir):
+            image_path = os.path.join(self.images_dir, image_file)
+            if os.path.isfile(image_path):
+                self.upload_image(image_path)
+                log.info(f"Uploaded {image_file} successfully.")
+
+    def upload_image(self, image_path):
+        url = f"{self.base_url}/task/new/upload/{self.task_uuid}"
```

Comment:
I haven't checked the API, but is there possibly and endpoint to upload a zip?

It might actually be more efficient to download all the images, zip, then upload in one go - worth assessing / checking !